spec/posts.spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  let(:user) { User.create(name: 'Oladayo', posts_counter: 0) }
+
+  subject do
+    Post.create(user_id: user.id, title: 'My first post', text: 'Here is my post.', comment_counter: 0, like_counter: 0)
+  end
+
+  let(:first_comment) do
+    Comment.create(post: subject, user:, text: 'first comment', created_at: 'Fri, 28 Jul 2023 04:51:46')
+  end
+  let(:second_comment) do
+    Comment.create(post: subject, user:, text: 'second comment', created_at: 'Fri, 28 Jul 2023 05:51:46')
+  end
+  let(:third_comment) do
+    Comment.create(post: subject, user:, text: 'third comment', created_at: 'Fri, 28 Jul 2023 06:51:46')
+  end
+  let(:fourth_comment) do
+    Comment.create(post: subject, user:, text: 'fourth comment', created_at: 'Fri, 28 Jul 2023 07:51:46')
+  end
+  let(:fifth_comment) do
+    Comment.create(post: subject, user:, text: 'fifth comment', created_at: 'Fri, 28 Jul 2023 08:51:46')
+  end
+  let(:sixth_comment) do
+    Comment.create(post: subject, user:, text: 'sixth comment', created_at: 'Fri, 28 Jul 2023 09:51:46')
+  end
+
+  let(:first_like) { Like.create(post: subject, user:) }
+  let(:second_like) { Like.create(post: subject, user:) }
+
+  shared_examples 'a non-negative integer' do |attribute|
+    it 'counter should not be a negative number' do
+      subject.send("#{attribute}=", -1)
+      expect(subject).to_not be_valid
+    end
+
+    it 'counter should be greater than or equal to zero' do
+      subject.send("#{attribute}=", 0)
+      expect(subject).to be_valid
+
+      subject.send("#{attribute}=", 1)
+      expect(subject).to be_valid
+    end
+
+    it 'counter should be an integer' do
+      subject.send("#{attribute}=", 1.5)
+      expect(subject).to_not be_valid
+    end
+  end
+
+  context '#validation' do
+    it 'should validate_presence_of(:title)' do
+      subject.title = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'should validate_length_of(:title).is_at_most(250)' do
+      subject.title = 'a' * 250
+      expect(subject).to be_valid
+
+      subject.title = 'a' * 251
+      expect(subject).to_not be_valid
+    end
+
+    it 'should validate_integer(:comment_counter)' do
+      subject.comment_counter = 1.7
+      expect(subject).to_not be_valid
+    end
+
+    it 'should validate_integer_greater_than_zero(:comment_counter)' do
+      subject.comment_counter = 1
+      expect(subject).to be_valid
+    end
+  end
```

Comment:
- [ ] Good job for covering most of the previous reviewer's suggestions. However, I think you missed one from the review summary.  The following hasn't been implemented, please make sure to address this 👇
  
  ![image](https://github.com/emarkees/Gist_Blog/assets/106119105/c2483f59-f8f1-4f1e-bace-88c2917947dd)

- [ ] Good job so far 👏👏 however, I cannot really see the tests of the `recent_comments` and `update_post_counter` methods present in the `post` model [as suggested](https://github.com/emarkees/Gist_Blog/pull/3#discussion_r1306577214), could you lease have a look at this ? You can try something like below 👇


```ruby
context '#callbacks' do
  it 'should increment the user\'s posts_counter after save' do
    expect {
      Post.create(user_id: user.id, title: 'My first post', text: 'Here is my post', comment_counter: 0, like_counter: 0)
      user.reload
    }.to change(user, :posts_counter).by(1)
  end
end

context '#recent_comments' do
  it 'should return the most recent comments' do
    first_comment
    second_comment
    third_comment
    fourth_comment
    fifth_comment
    sixth_comment

    recent_comments = subject.recent_comments

    expect(recent_comments.length).to eq(5)
    expect(recent_comments).to eq([sixth_comment, fifth_comment, fourth_comment, third_comment, second_comment])
  end
end
```

Reply:
@KanzaTahreem Thank you for the review. Changes made as suggested

Reply:
@emarkees I just sent you a message on Slack. Please take a look.